docs/programming-guide.md

```diff
@@ -1378,29 +1378,36 @@ res2: Long = 10
 
 While this code used the built-in support for accumulators of type Long, programmers can also
 create their own types by subclassing [AccumulatorV2](api/scala/index.html#org.apache.spark.util.AccumulatorV2).
-The AccumulatorV2 abstract class has several methods which need to override: 
-`reset` for resetting the accumulator to zero, and `add` for add anothor value into the accumulator, `merge` for merging another same-type accumulator into this one. Other methods need to override can refer to scala API document. For example, supposing we had a `MyVector` class
-representing mathematical vectors, we could write:
+The AccumulatorV2 abstract class has several methods which one has to override:
+`reset` for resetting the accumulator to zero, `add` for adding another value into the accumulator,
+and `merge` for merging another same-type accumulator into this one. Other methods that must be overridden
+are contained in the [API documentation](api/scala/index.html#org.apache.spark.util.AccumulatorV2).
+For example, supposing we had a `MathVector` class representing mathematical vectors, we could write:
```

Comment:
Your point makes sense to me. 

The renaming was to done to improve the consistency between Java/Scala examples. For instance, the Scala example used MyVector, while the Java one - Vector. 

Will it be a better idea to keep MyVector in the Scala example and reuse this name in Java?